src/apps/path-finder/store/search-thunk.ts

```diff
@@ -50,12 +50,12 @@ export function searchPath(
       cells.forEach((cell) => {
         grid[cell.row][cell.col] = cellType;
         visitedCellCount++;
+        dispatch(setVisitedCellCount(visitedCellCount));
       });
+      dispatch(setStateCells({ cells, cellType }));
```

Comment:
I believe this should be inside the condition below. We don't want to update the state every time if there is no delay set. Plz verify

Reply:
yess right